tests/test_flow/test_persistence.py

```diff
@@ -902,6 +902,69 @@ def wrapped_f(x, y):  # noqa: F811
     assert call_counter.times_called() == 1
 
 
+def test_versioning_manual_class_changes(builder, make_counter):
```

Comment:
Note: I ran this test on master to verify the bug and the fix. It fails on master whereas it passes with the changes in this PR.

Reply:
Should we test the assisted mode as well?